internal/adapter/job.go

```diff
@@ -0,0 +1,186 @@
+	containers, err := adapter.cli.ContainerList(ctx, types.ContainerListOptions{All: true, Filters: filter})
+	if err != nil {
+		return types.Container{}, fmt.Errorf("unable to list containers: %w", err)
+	}
+
+	if len(containers) == 0 {
+		return types.Container{}, adaptererr.ErrResourceNotFound
+	}
+
+	if len(containers) > 1 {
+		return types.Container{}, fmt.Errorf("multiple containers were found with the associated job name %s", jobName)
+	}
+
+	return containers[0], nil
+}
+
+func (adapter *KubeDockerAdapter) GetJob(ctx context.Context, jobName string, namespace string) (*batchv1.Job, error) {
+	container, err := adapter.getContainerFromJobName(ctx, jobName, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get container from job name: %w", err)
+	}
+
+	job, err := adapter.buildJobFromContainer(ctx, container)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get job: %w", err)
+	}
+
+	versionedJob := batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "batch/v1",
+		},
+	}
+
+	err = adapter.ConvertK8SResource(job, &versionedJob)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert internal object to versioned object: %w", err)
+	}
+
+	return &versionedJob, nil
+}
+
+func (adapter *KubeDockerAdapter) GetJobTable(ctx context.Context, namespace string) (*metav1.Table, error) {
+	jobList, err := adapter.listJobs(ctx, namespace)
+	if err != nil {
+		return &metav1.Table{}, fmt.Errorf("unable to list jobs: %w", err)
+	}
+
+	return k8s.GenerateTable(&jobList)
+}
+
+func (adapter *KubeDockerAdapter) ListJobs(ctx context.Context, namespace string) (batchv1.JobList, error) {
+	jobList, err := adapter.listJobs(ctx, namespace)
+	if err != nil {
+		return batchv1.JobList{}, fmt.Errorf("unable to list jobs: %w", err)
+	}
+
+	versionedJobList := batchv1.JobList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "JobList",
+			APIVersion: "batch/v1",
+		},
+	}
+
+	err = adapter.ConvertK8SResource(&jobList, &versionedJobList)
+	if err != nil {
+		return batchv1.JobList{}, fmt.Errorf("unable to convert internal JobList to versioned JobList: %w", err)
+	}
+
+	return versionedJobList, nil
+}
+
+func (adapter *KubeDockerAdapter) buildJobFromContainer(ctx context.Context, container types.Container) (*batch.Job, error) {
+	if container.Labels[k2dtypes.LastAppliedConfigLabelKey] == "" {
+		return nil, fmt.Errorf("unable to build job, missing %s label on container %s", k2dtypes.LastAppliedConfigLabelKey, container.Names[0])
+	}
+
+	jobData := container.Labels[k2dtypes.LastAppliedConfigLabelKey]
+
+	versionedJob := batchv1.Job{}
+
+	err := json.Unmarshal([]byte(jobData), &versionedJob)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal versioned job: %w", err)
+	}
+
+	job := batch.Job{}
+	err = adapter.ConvertK8SResource(&versionedJob, &job)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert versioned job spec to internal job spec: %w", err)
+	}
```

Comment:
Stuck on this line... I am able to start the job, but not list them, as it looks like there is no conversion func for `batchv1.Job` to `batch.Job`.

Reply:
```
2023-09-30T01:07:27.917+0200    DEBUG   middleware/logging.go:16        received HTTP request   {"url": "/apis/batch/v1/namespaces/default/jobs?limit=500", "method": "GET", "remote_address": "192.168.0.94:55322", "request_id": "ebb1e8e9-dc4c-4447-b334-5d8454fe4f70", "header_accept": "application/json;as=Table;v=v1;g=meta.k8s.io,application/json;as=Table;v=v1beta1;g=meta.k8s.io,application/json"}
2023-09-30T01:07:27.921+0200    ERROR   utils/response.go:43    unable to get table: unable to list jobs: unable to get job: unable to convert versioned job spec to internal job spec: converting (v1.JobSpec) to (batch.JobSpec): unknown conversion      {"request_id": "ebb1e8e9-dc4c-4447-b334-5d8454fe4f70"}
github.com/portainer/k2d/internal/api/utils.ListResources
        /Users/shanduur/Repos/k2d/internal/api/utils/response.go:43
github.com/portainer/k2d/internal/api/apis/batch/jobs.JobService.ListJobs
        /Users/shanduur/Repos/k2d/internal/api/apis/batch/jobs/list.go:14
github.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/filter.go:23
main.main.CheckAuthenticationHeader.func2
        /Users/shanduur/Repos/k2d/internal/middleware/authentication.go:25
github.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/filter.go:21
github.com/portainer/k2d/internal/middleware.LogRequests
        /Users/shanduur/Repos/k2d/internal/middleware/logging.go:24
github.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/filter.go:21
github.com/portainer/k2d/internal/middleware.AddTracingHeaders
        /Users/shanduur/Repos/k2d/internal/middleware/tracing.go:14
github.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/filter.go:21
main.main.func1
        /Users/shanduur/Repos/k2d/cmd/k2d.go:156
github.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/filter.go:21
github.com/emicklei/go-restful/v3.(*Container).dispatch
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/container.go:296
net/http.HandlerFunc.ServeHTTP
        /opt/homebrew/Cellar/go/1.21.1/libexec/src/net/http/server.go:2136
net/http.(*ServeMux).ServeHTTP
        /opt/homebrew/Cellar/go/1.21.1/libexec/src/net/http/server.go:2514
github.com/emicklei/go-restful/v3.(*Container).ServeHTTP
        /Users/shanduur/Repos/k2d/vendor/github.com/emicklei/go-restful/v3/container.go:316
net/http.serverHandler.ServeHTTP
        /opt/homebrew/Cellar/go/1.21.1/libexec/src/net/http/server.go:2938
net/http.initALPNRequest.ServeHTTP
        /opt/homebrew/Cellar/go/1.21.1/libexec/src/net/http/server.go:3546
net/http.(*http2serverConn).runHandler
        /opt/homebrew/Cellar/go/1.21.1/libexec/src/net/http/h2_bundle.go:6131
```

Reply:
Fixed, caused by missing adapter Schema extension.